python/grass/projpicker/common.py

```diff
@@ -0,0 +1,126 @@
+"""
+This module provides common variables and functions for other ProjPicker
+modules.
+"""
+
+import os
+import re
+import collections
+
+projpicker_verbose_env = "PROJPICKER_VERBOSE"
+
+# regular expression patterns
+# coordinate separator
+coor_sep = ","
+coor_sep_pat = f"[ \t]*[{coor_sep} \t][ \t]*"
+# positive float
+pos_float_pat = "(?:[0-9]+(?:\.[0-9]*)?|\.[0-9]+)"
+
+# bbox table schema
+bbox_schema = """
+CREATE TABLE bbox (
+    proj_table TEXT NOT NULL CHECK (length(proj_table) >= 1),
+    crs_name TEXT NOT NULL CHECK (length(crs_name) >= 2),
+    crs_auth_name TEXT NOT NULL CHECK (length(crs_auth_name) >= 1),
+    crs_code TEXT NOT NULL CHECK (length(crs_code) >= 1),
+    usage_auth_name TEXT NOT NULL CHECK (length(usage_auth_name) >= 1),
+    usage_code TEXT NOT NULL CHECK (length(usage_code) >= 1),
+    extent_auth_name TEXT NOT NULL CHECK (length(extent_auth_name) >= 1),
+    extent_code TEXT NOT NULL CHECK (length(extent_code) >= 1),
+    south_lat FLOAT CHECK (south_lat BETWEEN -90 AND 90),
+    north_lat FLOAT CHECK (north_lat BETWEEN -90 AND 90),
+    west_lon FLOAT CHECK (west_lon BETWEEN -180 AND 180),
+    east_lon FLOAT CHECK (east_lon BETWEEN -180 AND 180),
+    bottom FLOAT,
+    top FLOAT,
+    left FLOAT,
+    right FLOAT,
+    unit TEXT NOT NULL CHECK (length(unit) >= 2),
+    area_sqkm FLOAT CHECK (area_sqkm > 0),
+    CONSTRAINT pk_bbox PRIMARY KEY (
+        crs_auth_name, crs_code,
+        usage_auth_name, usage_code
+    ),
+    CONSTRAINT check_bbox_lat CHECK (south_lat <= north_lat)
+)
+"""
+
+# all column names in the bbox table
+bbox_columns = re.sub("^ +| +$", "",
+               re.sub("\n", " ",
+               re.sub("(?:^[A-Z]| ).*", "",
+               re.sub("\([^(]*\)", "",
+               re.sub("^(?:CREATE TABLE.*|\))$|^ *", "",
+                      bbox_schema, flags=re.MULTILINE),
+                      flags=re.DOTALL), flags=re.MULTILINE))).split()
+
+# BBox namedtuple class
+BBox = collections.namedtuple("BBox", bbox_columns)
+
+
+def is_verbose():
+    return os.environ.get(projpicker_verbose_env, "NO") == "YES"
+
+
+def get_float(x):
+    """
+    Typecast x into float; return None on failure.
+
+    Args:
+        x (str or float): Float in str or float.
+
+    Returns:
+        float or None: Typecasted x in float if successful, None otherwise.
+    """
+    if type(x) != float:
+        try:
+            x = float(x)
+        except:
```

Comment:
`except ValueError`

Reply:
In this specific case, I intend to catch *any* exceptions.

Reply:
I don't think your intention is to catch SystemExit and KeyboardInterrupt which is what bare except does. PEP 8 recommends against using bare except in its [Programming recommendations section](https://www.python.org/dev/peps/pep-0008/#programming-recommendations). Catching all (normal) exceptions is better expressed as `except Exception`, but the usual recommendation is to use more specific exception or exceptions which documents what you are expecting to happen and what you are not expecting to happen. This is partially because things like StopIteration and SyntaxError [inherit from Exception](https://docs.python.org/3/library/exceptions.html#exception-hierarchy). Another one in the list is NameError, so a wrong variable name (xl instead of x1) in the following code, still leads to the except part to be executed instead of Python reporting that xl does not exist. 

```
x1 = "5"
try:
    float(xl)
except Exception:
    print("All is fine")
```

What are you expecting besides invalid strings (ValueError)? Types which can't be converted to float (TypeError)? An integer (`int`) which is too large to convert ([OverflowError](https://docs.python.org/3/library/functions.html?highlight=float#float))?

Reply:
Makes sense, but I was not able to find the list of errors that `float()` can raise. `help(float)` only mentions `OverflowError` and `ValueError` from its `as_integer_ration()` member function. Curious if `float()` can even throw `SystemExit`, `KeyboardInterrupt`, `StopIteration`, or `SyntaxError` or, in other words, that `try` `except` can catch those when it only encloses a `float()`?

Reply:
For now, catches `ValueError` only.